mantle/cmd/ore/gcloud/promote-image.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2020 Red Hat
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gcloud
+
+import (
+	"github.com/coreos/mantle/platform/api/gcloud"
+	"github.com/spf13/cobra"
+	"golang.org/x/net/context"
+)
+
+var (
+	cmdPromoteImage = &cobra.Command{
+		Use:   "promote-image",
+		Short: "Promote GCP image in image family",
+		Long:  "Promote GCP image in image family and deprecate all others",
+		Run:   runPromoteImage,
+	}
+
+	promoteImageName   string
+	promoteImageFamily string
+)
+
+func init() {
+	cmdPromoteImage.Flags().StringVar(&promoteImageName, "image", "", "GCP image name")
+	cmdPromoteImage.Flags().StringVar(&promoteImageFamily, "family", "", "GCP image family to promote within")
+	GCloud.AddCommand(cmdPromoteImage)
+}
+
+func deprecateImage(name string, state gcloud.DeprecationState, replacement string) {
+	plog.Infof("Changing deprecation state of image: %v -> %v",
+		name, string(state))
+	pending, err := api.DeprecateImage(name, state, replacement)
+	if err == nil {
+		err = pending.Wait()
```

Comment:
In principle we should collect up all the pendings and wait on them at the end.  I don't know how much it matters here, though, since we're probably only deprecating one image.

Reply:
right this should be one API call IIUC